includes/class-wc-payments-order-service.php

```diff
@@ -520,6 +520,36 @@ public function get_charge_id_for_order( $order ) : string {
 		return $order->get_meta( self::CHARGE_ID_META_KEY, true );
 	}
 
+	/**
+	 * Get the dispute id from an order's notes if it exists.
+	 *
+	 * @param  mixed $order The order Id or order object.
+	 *
+	 * @return string|null
+	 *
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function get_dispute_id_for_order( $order ) : ?string {
+		$order = $this->get_order( $order );
+		// Get current notes of the order.
+		$current_notes = wc_get_order_notes(
+			[ 'order_id' => $order->get_id() ]
+		);
+
+		foreach ( $current_notes as $current_note ) {
+			if ( $current_note->content ) {
+				$matches = [];
+				// Regex pattern to match dispute id e.g. "dp_1NpKDUC15a3WbJ1tRUP8UorZ".
+				preg_match( '/dp_[a-zA-Z0-9]{24}/', $current_note->content, $matches );
```

Comment:
This is an initial idea of how to find the dispute ID – there may be an opportunity to make this more robust and/or more performant.

Reply:
This won't work after we remove the dispute details page as the link will go to the transaction details page, so no `dp_` won't be found.

Reply:
Good catch!